homeassistant/components/zeroconf/__init__.py

```diff
@@ -553,11 +553,19 @@ def info_from_service(service: AsyncServiceInfo) -> ZeroconfServiceInfo | None:
             break
     if not host:
         return None
+
+    service_properties = service.properties
+    if TYPE_CHECKING:
+        # Service properties are always bytes if they are set from the network.
+        # For legacy backwards compatibility zeroconf allows properties to be set
+        # as strings but we never do that so we can safely cast here.
+        service_properties = cast(dict[bytes, bytes | None], service_properties)
```

Comment:
This was typed to `dict` before in upstream `zeroconf` so `mypy` never errored before